octodns/record/__init__.py

```diff
@@ -80,6 +84,20 @@ def __init__(self, fqdn, reasons):
 class Record(EqualityTupleMixin):
     log = getLogger('Record')
 
+    _CLASSES = {}
+
+    @classmethod
+    def register_type(cls, _class, _type=None):
+        if _type is None:
+            _type = _class._type
+        existing = cls._CLASSES.get(_type, None)
```

Comment:
Minor cosmetics: any value in explicitly setting default fallback to `None`?

Reply:
If i follow the question, the `None` default prevents this line from throwing a `KeyError` exception when the class isn't already registered. This could have been done with a try/except if that's preferable, e.g.

```python
        try:
            existing = cls._CLASSES[_type]
            module = existing.__module__
            name = existing.__name__
            msg = f'Type "{_type}" already registered by {module}.{name}'
            raise RecordException(msg)
        except KeyError:
            cls._CLASSES[_type] = _class
```

Reply:
Oh no I meant `.get(_type, None)` vs just `.get(_type)`

Reply:
Ah, maybe I started out with `False` in there or something. I wasn't even seeing that bit.

Reply:
Like I said, minor cosmetic, no big deal